src/components/TodoList/TodoList.tsx

```diff
@@ -1,12 +1,57 @@
-/* eslint-disable max-len */
-import React from 'react';
+import classNames from 'classnames';
+import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../../app/hooks';
+import { actions } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
 
 export const TodoList: React.FC = () => {
+  const dispatch = useDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const { query, status } = useAppSelector(state => state.filter);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([...todos]);
+
+  function sortTodos() {
+    let correctListOfTodos;
+
+    switch (status) {
+      case 'active':
+        correctListOfTodos = [...todos].filter(todo => !todo.completed);
+        break;
+
+      case 'completed':
+        correctListOfTodos = [...todos].filter(todo => todo.completed);
+        break;
+
+      default:
+        correctListOfTodos = [...todos];
+        break;
+    }
+
+    setVisibleTodos(correctListOfTodos.filter(todo => (
+      todo.title.toLowerCase()).includes(query.toLowerCase())));
+  }
+
+  useEffect(() => {
+    sortTodos();
+  }, [query, status]);
```

Comment:
Would be better to wrap the function into the useCallback hook and use it in the dependency array. Fix it, please

```suggestion
  useEffect(() => {
    sortTodos();
  }, [sortTodos]);
```

Reply:
Круте рішення, дякую за підказку)